luaui/Widgets/cmd_nano_range_check.lua

```diff
@@ -0,0 +1,277 @@
+local function processNewOrders()
+    if #new_orders == 0 and #to_be_cleared > 0 then return end
+    for i = #new_orders, 1, -1 do
+        if command_budget <= 0 then break end
+        Spring.GiveOrderArrayToUnit(new_orders[i][1], new_orders[i][2])
+        table.remove(new_orders, i) -- pop
+        command_budget = command_budget - 1
+    end
+end
+
+-- Processes the turret range check after a certain number of frames (controlled by `DELAY`).
+-- It iterates over the current list of nano turrets and checks their range to ensure they are not assigned
+-- to constructions out of reach.
+-- If any changes are detected, it schedules new orders or/and clears existing ones.
+local function processTurretRange()
+    if counter >= DELAY then
+        counter = 0
+        for _, tower in ipairs(current_towers) do
+            checkTurretRange(tower)
+        end
+        flag_processed = true
+    end
+end
+
+-- Handles the flow control. If no command is scheduled, it disables the listener for the gameframe loop,
+-- else it keeps calling the `processTurretRange` function, in case anything changed during re-ordering.
+local function handleFlowControl()
+    if #to_be_cleared == 0 and #new_orders == 0 then
+        if flag_processed then
+            flag_listening = false
+        else
+            processTurretRange()
+        end
+    end
+end
+-- !SECTION Main functions
+
+-- SECTION Widget functions
+function widget:Initialize()
+    is_play = Spring.GetSpectatingState()
+    -- pre-check set
+    for _, name in ipairs(TURRETS) do
+        if UnitDefNames[name].buildDistance == nil then
+            Spring.Echo("Error: " .. name .. " has no buildDistance")
+            widget:Shutdown()
+        end
+    end
+    tableToSet(TURRETS)
+end
+
+function widget:GameFrame()
+    if flag_listening then -- only as long as turret is selection and one post-frame after selection drop.
+        counter = counter + 1
+        command_budget = COMMAND_LIMIT
+
+        applyStopOrders()
+        processNewOrders()
+        handleFlowControl()
+    end
+end
+
+-- Grabs any nano in selection and stores it in the list
+function widget:SelectionChanged(selectedUnits)
+    if is_play ~= false then return end
+    if flag_processed == true then -- remove old when processing is done, else append.
+        current_towers = {}
+    end
+    for _, unitID in ipairs(selectedUnits) do
+        local unitDefID = Spring.GetUnitDefID(unitID)
+        if TURRETS[UnitDefs[unitDefID].name] then
+            table.insert(current_towers, unitID)
+        end
+    end
+end
+
+-- Listener to when nano receives a command
+function widget:UnitCommand(uID, _, _, _, _, _, _)
+    if is_play ~= false then return end
+    for _, nano in ipairs(current_towers) do
+        if uID == nano then
+            flag_listening = true
+            counter = 0
+            flag_processed = false
+            break
+        end
+    end
+end
```

Comment:
Why not just check the unitDefId against the list of valid unit def IDs? Much shorter list to iterate through rather than all current individual units.